students/BrandonHenson/Lesson9/Mailroom_5.py

```diff
@@ -0,0 +1,151 @@
+# Brandon Henson
+# 5/20/18
+# Lesson 09 mailroom_5.py
+# Fixing issues:Refactoring
+# !/usr/bin/env python3
+
+
+class Donor():
+    def __init__(self, name=None):
+        self._name = name
+        self._donations = []
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, val):
+        self._name = val
+
+    @property
+    def donations(self):
+        return self._donations
+
+    @donations.setter
+    def donations(self, val):
+        self._donations = val
+
+    @property
+    def total(self):
+        return sum(self.donations)
+
+
+class DonorAll(dict):
+    def __init__(self, donors):
+        self._donors = donors
+
+    @property
+    def donors(self):
+        return self._donors
+
+    @donors.setter
+    def donors(self, val):
+        self._donors = val
+
+    def add_donor(self, donor):
+        self.donors[donor.name] = donor.donations
```

Comment:
Great choice to use a dictionary for easy search functionality to find your donor, however I would recommend value to be a Donor object so you can access anything you want about donor.